### PR TITLE
Adapt teuthology-openstack to latest state of the world

### PR DESCRIPTION
**Pull request by smithfarm**
This aspires to address https://bugzilla.suse.com/show_bug.cgi?id=1023257 and https://bugzilla.suse.com/show_bug.cgi?id=1023263

The fix for the latter bug is entirely written by Loic Dachary.

### CODE REVIEW

**Comment by smithfarm** [comment]
Hm, so now after it builds RPMs it says:

```
2017-02-02T22:43:19.237 INFO:tasks.buildpackages:Finished buildpackages
2017-02-02T22:43:19.238 INFO:teuthology.run_tasks:Running task internal.lock_machines...
2017-02-02T22:43:19.240 INFO:teuthology.task.internal:Locking machines...
2017-02-02T22:43:19.245 DEBUG:teuthology.report:Pushing job info to http://localhost:8080/
2017-02-02T22:43:19.396 INFO:teuthology.task.internal:waiting for more openstack machines to be free (need 5 + 1, have 0)...
2017-02-02T22:43:29.440 INFO:teuthology.task.internal:waiting for more openstack machines to be free (need 5 + 1, have 0)...
2017-02-02T22:43:39.465 INFO:teuthology.task.internal:waiting for more openstack machines to be free (need 5 + 1, have 0)...
```

whereas before it said:

```
2017-02-02T20:07:42.939 INFO:tasks.buildpackages:Finished buildpackages
2017-02-02T20:07:42.939 INFO:teuthology.run_tasks:Running task internal.lock_machines...
2017-02-02T20:07:42.942 INFO:teuthology.task.internal:Locking machines...
2017-02-02T20:07:42.943 DEBUG:teuthology.report:Pushing job info to http://localhost:8080/
2017-02-02T20:07:43.031 DEBUG:teuthology.provision:ProvisionOpenStack: {'subnet': '167.114.224.0/19', 'user-data': 'teuthology/openstack/openstack-{os_type}-{os_version}-user-data.txt', 'ip': '167.114.253.177', 'clone': 'git clone http://github.com/ceph/teuthology', 'machine': {'disk': 20, 'ram': 8000, 'cpus': 1}, 'volumes': {'count': 0, 'size': 1}, 'flavor-select-regexp': '^(vps|hg)-.*ssd', 'nameserver': '167.114.253.177'}
2017-02-02T20:07:43.031 DEBUG:teuthology.provision:ProvisionOpenStack:create
```

Will look at that more tomorrow.

**Comment by smithfarm** [comment]
Messages from `setup-openstack.sh --setup-all` are logged to `/tmp/init.out` - this could be moved to `/var/log/setup-openstack.log` by an init script as part of the boot process?

**Comment by smithfarm** [comment]
```
INSTALLED dnsmasq and configured to be a resolver
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/ubuntu/teuthology/virtualenv/local/lib/python2.7/site-packages/netaddr/ip/__init__.py", line 933, in __init__
    raise AddrFormatError('invalid IPNetwork %s' % addr)
netaddr.core.AddrFormatError: invalid IPNetwork teuthology
etc.
 * Stopping DNS forwarder and DHCP server dnsmasq
   ...done.
 * Starting DNS forwarder and DHCP server dnsmasq
   ...done.
CREATED /etc/dnsmasq.d/teuthology
```

**Comment by smithfarm** [comment]
Now the output is:

```
Configuring dnsmasq for subnets 
etc.
 * Stopping DNS forwarder and DHCP server dnsmasq
   ...done.
 * Starting DNS forwarder and DHCP server dnsmasq
   ...done.
CREATED /etc/dnsmasq.d/teuthology
```

So, `subnets` is empty and hence `/etc/dnsmasq.d/teuthology` is empty as well.

**Comment by dachary** [comment]
@smithfarm did you bash -x the script to get the details about what it is doing ?

File: teuthology/openstack/setup-openstack.sh [hunk 1]
```diff
@@ -605,11 +615,13 @@ function main() {
     #
     # assume the first available IPv4 subnet is going to be used to assign IP to the instance
     #
-    eval local default_subnet=$(neutron subnet-list -f json -c cidr -c ip_version | jq '.[] | select(.ip_version == 4) | .cidr' | head -1)
-    if test -z "$default_subnet" ; then
-        default_subnet=$(nova tenant-network-list | grep / | cut -f6 -d' ' | head -1)
+    local default_subnets=$(neutron subnet-list -f json -c cidr -c ip_version | jq '.[] | select(.ip_version == 4) | .cidr')
+    if test -z "$default_subnets" ; then
```
Comment (jan--f) [inline teuthology/openstack/setup-openstack.sh]: This test is the culprit...should b -n

**Comment by jan--f** [comment]
pushed a tiny fix and testing it now.

**Comment by smithfarm** [comment]
Loic's original, for posterity: http://paste2.org/9V8gKXZZ

**Comment by dachary** [comment]
@jan--f I think the -z is right: you want that only when the user did not --subnets. Or am I missing something ?

**Comment by jan--f** [comment]
Before my commit the check did "aggregate $default_subnets in $subnets if $default_subnets is empty". I changed it to "if not empty".

is there may be a "test -z $subnets" missing?

**Comment by smithfarm** [comment]
`DEBUG:teuthology.provision:ProvisionOpenStack: {'subnet': '213.32.72.0/21 149.202.160.0/19 137.74.24.0/21', 'user-data': 'teuthology/openstack/openstack-{os_type}-{os_version}-user-data.txt', 'ip': '213.32.73.203', 'clone': 'git clone http://github.com/ceph/teuthology', 'machine': {'disk': 20, 'ram': 8000, 'cpus': 1}, 'volumes': {'count': 0, 'size': 1}, 'flavor-select-regexp': '^(vps|hg)-.*ssd', 'nameserver': '213.32.73.203'}`

It's working now, but I'm a little worried about `'subnet': '213.32.72.0/21 149.202.160.0/19 137.74.24.0/21'`

**Comment by dachary** [comment]
> little worried about 'subnet': '213.32.72.0/21 149.202.160.0/19 137.74.24.0/21'

I was as well but it turns out this is not used anywhere. Unless I missed something which is possible...

**Pull request merged**